apis/refs/v1beta1/interface.go

```diff
@@ -92,3 +92,7 @@ func Normalize(ctx context.Context, reader client.Reader, ref Ref, defaultNamesp
 
 	return ref.ValidateExternal(ref.GetExternal())
 }
+
+type RefinerWithProjectID interface {
```

Comment:
Why we need this interface rather than use existing `Ref` or ` ExternalNormalizer` that normalize the reference?
* Because the projectID comes from the caller function `refNormalize`,  that it cannot be passed into the normalize function (If we do, we need  to change the other ~200 references)  

Why we cannot keep the special logic inside `NormalizeReference` function (as it is before) but use a new interface? 
* Because otherwise we cannot extend  the function NormalizeReference to other existing 200 resources (and the future 800 resources) which implements the `ExternalNormalizer` interface, since the monitoring will break (its normalize function breaks its own refine function as the interface level Normalize is called)

Reply:
So no objection to a second interface. 

* I think the function should be named NormalizeReference<foo>, not Refine<foo>
* I think the function should take a NormalizeReferenceOptions struct, for extensibility.  So maybe NormalizeReferenceWithOptions?

BTW I don't think it's a huge deal to fix these interfaces (I think it's a search and replace with some regex), but I also have no problem with doing it more incrementally by introducing a V2 interface.